packages/xrpl/test/integration/onConnect.test.ts

```diff
@@ -12,13 +12,11 @@ describe('on handlers', function () {
     'on connect',
     async () => {
       const client = new Client(serverUrl)
-      return new Promise<void>(function (resolve) {
-        client.on('connected', function () {
-          client.removeAllListeners()
-          client.disconnect().then(() => resolve())
-        })
-        client.connect()
+      client.on('connected', function () {
```

Comment:
It is because once connected client.connect() also calls server_info to retrieve networkId and buildVersion, which would takes longer than client.disconnect(), hence will cause the error since client has already been disconnected while connect() is still running so I have to move client.disconnected() outside of the on connected function

Reply:
@pdp2121 what error is caused? Can you give more information?

This leads me to believe your code needs to be enhanced to handle the case where the client is disconnected after you've already called server_info --- or that when your call to server_info is ready to be made, the client could already be disconnected.

I'm curious to know the error before we call this resolved

Reply:
`xrpl:     DisconnectedError: WebSocket is not open: readyState 2 (CLOSING)
xrpl: 
xrpl:       161 |     ws.send(message, (error) => {
xrpl:       162 |       if (error) {
xrpl:     > 163 |         reject(new DisconnectedError(error.message, error))
xrpl:           |                ^
xrpl:       164 |       } else {
xrpl:       165 |         resolve()
xrpl:       166 |       }`

Reply:
The error is as above. Talked with @JST5000 about this and we decided to move `disconnect()` outside of `on('connected'...)`

Reply:
In practice we would not put `disconnect()` inside connected listener anyway so everything inside connected will resolve first and the issue would not happen

Reply:
It sounds like this is a bug in your code is what I'm saying. Where is this error being thrown from? Is it inside the new code you added to the .then ?

Reply:
@justinr1234 "connected" will emit when the websocket is connected to server, and after that the code execute the `server_info` call. For the previous test to work (`disconnect()` happens immediately on "connected"), the `server_info` call has to finish its execution before the websocket emits "connected" message. Are you suggesting we only emit "connected" after the `server_info` call finish its execution? @ckniffen @mvadari @JST5000 please let me know what you think as well. Thanks!

Reply:
I would suggest that we only emit "connected" after the `server_info` call finishes it's connection. My understanding is that as soon as `connected` finishes, we can expect the user to create transactions and expect them to work. Since the information from `server_info` will be required to `autofill` a transaction for some networks, I believe you should have to wait until that's done beforehand.

Reply:
That makes sense. Yea I can work on that. Thanks @justinr1234 @JST5000

Reply:
It’s my understanding that server info isn’t necessary and is more of an optional thing. Unless I’m mistaken.

I am not saying we should wait for server info. The opposite, I’m suggesting we should allow the current behavior to continue to work